miner/worker.go

```diff
@@ -318,6 +318,8 @@ func (w *worker) isRunning() bool {
 // close terminates all background threads maintained by the worker.
 // Note the worker does not support being closed multiple times.
 func (w *worker) close() {
+	w.mu.Lock()
+	defer w.mu.Unlock()
```

Comment:
Btw, I think one better fix is to move the `w.current.state.StopPrefetcher()` operation into the `mainLoop`.  Since `w.current` is only available in this loop and you can tear down it in the defer function if the worker is stopped.

Reply:
I looked into this a bit more... 
The `worker.close` is only called from `miner.Close`. I renamed it like this: 
```
func (miner *Miner) zClose() {
	close(miner.exitCh)
}
```
And geth still compiles. So I'm wondering, do we actually _ever_ call `miner.Close() -> worker.close()` ?